src/core/connection.h

```diff
@@ -463,6 +463,21 @@ typedef struct QUIC_CONNECTION {
     //
     uint64_t NextRecvAckFreqSeqNum;
 
+    //
+    // The maximum number of packets that can be out of order before an immediate
+    // acknowledgment (ACK) is triggered. If no specific instructions (ACK_FREQUENCY
+    // frames) are received from the peer, the receiver will immediately acknowledge
+    // any out-of-order packets, which means the default value is 1. A value of 0
+    // means out-of-order packets do not trigger an immediate ACK.
+    //
+    uint8_t ReorderingThreshold;
+
+    //
+    // The maximum number of packets that the peer can be out of order before an immediate
+    // acknowledgment (ACK) is triggered.
+    //
+    uint8_t PeerReorderingThreshold;
```

Comment:
Is this meant to be what we report to the peer, or what they report to us? Because the code you have seems to be doing both. You're initializing this to our own default, setting it when you receive a frame from the peer and using it when you send frames to the peer. It shouldn't be both. Likely, it should just be what the peer sends to us. For the frames we send, we can hard code something for now.

Reply:
When we receive packets, I am setting ReorderingThreshold not PeerReorderingThreshold. PeerReorderingThreshold is supposed to be meant for what we report to the peer while ReorderingThreshold is what they report to us.

Reply:
Ah, you're right. My fault. Sorry!